external-crates/move/crates/move-compiler/src/typing/visitor.rs

```diff
@@ -163,24 +239,156 @@ pub trait TypingVisitorContext {
             self.pop_warning_filter_scope();
             return;
         }
+        if Self::VISIT_TYPES {
+            fdef.signature
+                .parameters
+                .iter_mut()
+                .map(|(_, _, ty)| ty)
+                .for_each(|ty| self.visit_type(ty));
+            self.visit_type(&mut fdef.signature.return_type);
+        }
         if let T::FunctionBody_::Defined(seq) = &mut fdef.body.value {
             self.visit_seq(seq);
         }
         self.pop_warning_filter_scope();
     }
 
-    fn visit_seq(&mut self, (_, seq): &mut T::Sequence) {
+    // -- TYPES --
+
+    fn visit_type_custom(&mut self, _ty: &mut N::Type) -> bool {
+        false
+    }
+
+    /// Visit a type, including recursively. Note that this may be called manually even if
+    /// `VISIT_TYPES` is set to `false`.
+    #[growing_stack]
```

Comment:
When do we need to include the `#[growing_stack]` annotations? Is it related to recursion (though it's not specified for `visit_lvalue` so probably not...)?

Reply:
We generally do when recurring. it should be on `visit_lvalue`.